policybot/pkg/testresults/testresults.go

```diff
@@ -0,0 +1,337 @@
+
+/**
+ * Take in a pr number from path "istio-prow/pr-logs/pull/istio-istio" and examine the pr
+ * for all tests that are run and their results. The results are then written to Spanner.
+ */
+package testresults
+
+import (
+	"context"
+	"encoding/json"
+
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+
+	store "istio.io/bots/policybot/pkg/storage"
+	"istio.io/pkg/log"
+)
+
+/*
+ * Pull struct for the structure under refs/pulls in clone-records.json
+ */
+type pull struct {
+	Number int
+	Author string
+	Sha    string
+}
+
+/*
+ * Cmd struct for Commands object under clone-records.json
+ */
+type cmnd struct {
+	Command string
+	Output  string
+}
+
+/*
+ * Finished struct to store values for all fields in Finished.json
+ */
+type finished struct {
+	Timestamp int64
+	Passed    bool
+	Result    string
+}
+
+/*
+ * Clone_Record struct to store values for all fields in clone-records.json
+ */
+type cloneRecord struct {
+	Refs struct {
+		Org       string
+		Repo      string
+		BaseRef   string `json:"base_ref"`
+		BaseSha   string `json:"base_sha"`
+		Pulls     []pull
+		PathAlias string
+	}
+	Commands []cmnd
+	Failed   bool
+}
+
+/*
+ * Started struct to store values from started.json
+ */
+type started struct {
+	Timestamp int64
+}
+
+type PrResultTester struct {
+	Client     *storage.Client
+	ctx        context.Context
+	bucketName string
+}
+
+var scope = log.RegisterScope("TestResult", "Check error while reading from google cloud storage", 0)
+
+/*
+ * Function to initionalize new PrFlakeTest object.
+ * Use `prFlakeyTest, err := NewPrFlakeTest()` and `testFlakes := prFlakeyTest.checkTestFlakesForPr(<pr_number>)`
+ * to get test flakes information for a given pr.
+ */
+func NewPrResultTester(ctx context.Context, bucketName string) (*PrResultTester, error) {
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &PrResultTester{
+		Client:     client,
+		ctx:        ctx,
+		bucketName: bucketName,
+	}, nil
+}
+
+func (prt *PrResultTester) query(prefix string) ([]string, error) {
+	ctx := prt.ctx
+	client := prt.Client
+	bucket := client.Bucket(prt.bucketName)
+	query := &storage.Query{Prefix: prefix, Delimiter: "/"}
+	it := bucket.Objects(ctx, query)
+	paths := []string{}
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+		paths = append(paths, attrs.Prefix)
+
+	}
+	return paths, nil
+}
+
+/*
+ * GetTest function get all directories under the given pr in istio-prow/pr-logs/pull/istio-istio/PRNUMBER for each test suite name.
+ * Client: client used to get buckets and objects.
+ * PrNum: the PR number inputted.
+ * Return []Tests return a slice of Tests objects.
+ */
+func (prt *PrResultTester) getTests(prNumInt int64) (map[string][]string, error) {
+	prNum := strconv.FormatInt(prNumInt, 10)
+	prefixForPr := "pr-logs/pull/istio_istio/" + prNum + "/"
```

Comment:
The path generated here needs to take into consideration the org and repo and not hardcode those. So this needs to be `/pull/<org_login>_<repo_name>`...